src/Foundatio.AzureStorage/Storage/AzureFileStorage.cs

```diff
@@ -57,20 +57,19 @@ public async Task<Stream> GetFileStreamAsync(string path, StreamMode streamMode,
         if (String.IsNullOrEmpty(path))
             throw new ArgumentNullException(nameof(path));
 
-        if (streamMode is StreamMode.Write)
-            throw new NotSupportedException($"Stream mode {streamMode} is not supported.");
-
         string normalizedPath = NormalizePath(path);
         _logger.LogTrace("Getting file stream for {Path}", normalizedPath);
 
         var blockBlob = _container.GetBlockBlobReference(normalizedPath);
 
         try
         {
-            return await blockBlob.OpenReadAsync(null, null, null, cancellationToken).AnyContext();
-        }
-        catch (StorageException ex) when (ex is { RequestInformation.HttpStatusCode: 404 })
-        {
+            return streamMode switch {
+                StreamMode.Read => await blockBlob.OpenReadAsync(null, null, null, cancellationToken).AnyContext(),
+                StreamMode.Write => await blockBlob.OpenWriteAsync(null, null, null, cancellationToken).AnyContext(),
+                _ => null
```

Comment:
We might want to throw here in case we add a new mode.
```suggestion
                _ => throw new NotSupportedException($"Stream mode {streamMode} is not supported.");
```